### DISCUSSION TITLE
Is it possible to patch variable version of font?

### DISCUSSION THREAD

**Discussion by karambaq** [ryanoasis/nerd-fonts#1154] [Q&A]
I like the variable version of this font because it has `Semibold` option while usual is not. 
https://github.com/iaolo/iA-Fonts/tree/master/iA%20Writer%20Mono/Variable

However, when I'm patching this font font-patcher gives me an warning:

```
Warning: Source font is a variable open type font (VF) and the patch results will most likely not be what you want
   iM Writing Mono V Nerd Font
   \===> 'iM Writing Mono V Nerd Font.ttf'
```

And after patching and adding this font to the system it only have 1 weight available while has 8 by default. 

Any suggestions would be great!

**Comment by Finii** [comment]
I _believe_ it is not really possible to work with variable fonts in `fontforge`. And the Nerd Fonts `font-patcher` is Fontforge script.

Frederick (@ctrlcctrlv) shows in this video how one can use fontforge to work on VF, with additional tools:
https://www.youtube.com/watch?v=xoQuWARCUWI

But still, a **generic** way to patch VF needs to take into account a number of thing as there can be not only two but more masters in a VF even if you just have one axis. Which makes that a rather complex project. I mean expanding Nerd Fonts to support VF generically. On the other hand, maybe you just (blindly) patch in the same outline into each master?
One would need to work on that, and it will take quite some effort, I guess.

But to solve your problem: One could just instantiate a static font from your VF with the weight you want. And then patch that static font. That is at least a solution. You then install the weights you need and work with un-variable fonts that have concrete weights. I guess a terminal emulator application can not utilize the real _variable_ in VF anyhow and just allows to use the predefined weights that there are.

I could create you a static instance of your desire, if you like and if that is a solution for you. I would just open the VF in `Glyphs` and export the instance, which is a 2 minute job. I would need the concrete weight and link to the font file (which you probably gave above, but please repeat in the answer if you like me to do that).
There are open source tools to do that _I guess_, but I never needed them, so I have no suggestions.

**Reply by ctrlcctrlv** [reply]
You're incorrect. See my answer. I don't know why the symbols would have to vary, even assuming patching symbols into fonts makes sense. You can just have the points be in the same location at both ends of the axis.

**Comment by karambaq** [comment]
Hey @Finii!

Thanks for you answer, I think I'll just move to another font then :)

**Comment by ctrlcctrlv** [comment]
Avoid FontForge entirely for this task.

My video is not the best way to do this.

What you want to do is define your glyphs as static .glif's (as your symbols should not vary) then use `ufo-extractor` Python library to make the binary into source UFO:

```python3
>>> import extractor
>>> import defcon
>>> ufo = defcon.Font()
>>> extractor.extractUFO("/path/to/MyFont.ttf", ufo)
>>> ufo.save("/path/to/MyFont.ufo")
```

Then copy your .glif files into **every master**, update their contents directories e.g. with MFEKmetadata or by adding the .glif's through a Python library, and then `fontmake`.

**Reply by Finii** [reply]
Thanks for the information!

(Although I do not see how my answer is incorrect. What you write here is the same 'put the icon glyphs into every master' that I wrote. But maybe some other aspect is wrong.)

**Reply by ctrlcctrlv** [reply]
Ahha. I meant it in regards to:

> Which makes that a rather complex project.

I don't think it is one :-)
